src/components/SingleArticle.jsx

```diff
@@ -0,0 +1,18 @@
+import { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { makeSingleArticle } from "../../api-calls/api-processing";
+import "../../styling/SingleArticle.css"
+
+function SingleArticle() {
+  const [singleArticle, setSingleArticle] = useState(<></>);
+  const { article_id } = useParams();
+  useEffect(() => {
+    makeSingleArticle(article_id).then((articleCard) => {
```

Comment:
Interesting approach -  quite like the notion of creating the component as part of the api call. 
I think this reduces the overall readability of each component, but its a different approach  and it stands out!